core/templates/dev/head/pages/collection_player/CollectionPlayer.js

```diff
@@ -244,5 +242,11 @@ oppia.controller('CollectionPlayer', [
           'There was an error loading the collection.');
       }
     );
+
+    $scope.$watch('collection', function(newValue) {
```

Comment:
@seanlip -- I tried a simple direct call, but 'collection' isn't initialized (has a null value) at the time of call and results in errors. So, I added a $watch to call the function only when collection value is not null.